lib/scenic/adapters/postgres.rb

```diff
@@ -153,11 +155,49 @@ def update_materialized_view(name, sql_definition)
         raise_unless_materialized_views_supported
 
         IndexReapplication.new(connection: connection).on(name) do
-          drop_materialized_view(name)
-          create_materialized_view(name, sql_definition)
+          replace_materialized_view(name, sql_definition)
         end
       end
 
+      # Replaces a materialized view in the database.
+      #
+      # Recreate the materialized view under a temporary name and apply new
+      # indexes (if any). Drop previous materialized view, renames the
+      # temporary materialized view to its original name.
+      #
+      # This is typically called in a migration via {Statements#replace_view}.
+      #
+      # @param name The name of the view to update
+      # @param sql_definition The SQL schema for the updated view.
+      #
+      # @raise [MaterializedViewsNotSupportedError] if the version of Postgres
+      #   in use does not support materialized views.
+      #
+      # @return [void]
+      def replace_materialized_view(name, sql_definition)
+        raise_unless_materialized_views_supported
+
+        temp_mv_name = "#{name}_temp"
+
+        mv_definition = get_definition_for(:create_definition, sql_definition)
+        create_materialized_view(temp_mv_name, mv_definition.first)
+
+        new_index_definitions = get_definition_for(:indexes, sql_definition)
```

Comment:
We do not support specifying indexes in the SQL file. I see you've added that here, but I'd prefer to have people use `add_index` in migrations for this. I do not want to duplicate what can already be done with AR migrations.